suggestion_jobs.go

```
@@ -0,0 +1,323 @@
+	store.Lock()
+	defer store.Unlock()
+	if job, exists := store.jobs[jobID]; exists && job.Status == "pending" {
+		job.Status = "cancelled"
+		job.Error = "Job cancelled by user"
+		job.UpdatedAt = time.Now()
+		return true
+	}
+	return false
+}
+
+func (store *SuggestionJobStore) isCancelled(jobID string) bool {
```

golangci-lint (2.12.2)

[error] func (*SuggestionJobStore).isCancelled is unused

(unused)